pkg/xds/context/context.go

```diff
@@ -66,11 +66,16 @@ type MeshContext struct {
 	CrossMeshEndpoints     map[xds.MeshName]xds.EndpointMap
 	VIPDomains             []xds.VIPDomains
 	VIPOutbounds           []*mesh_proto.Dataplane_Networking_Outbound
-	ServiceTLSReadiness    map[string]bool
+	ServiceInformations    ServiceInformations
 	DataSourceLoader       datasource.Loader
 	ReachableServicesGraph ReachableServicesGraph
 }
 
+type ServiceInformations struct {
+	TLSReadiness map[string]bool
+	Protocol     map[string]core_mesh.Protocol
+}
```

Comment:
Would it make more sens to have ServiceInformations `map[string]ServiceInformation` instead of the double map?

Reply:
I think we can change it, for storing the data it might be better but using structure would require some validations

Reply:
Why required validation?

Reply:
Why you are there should you also add `isExternalService` ? Feels like it was the other things required to reduce our reliance on `core_xds.Routing` in plugins.

Reply:
The one issue that I noticed is circular dependency I might need to change some packages

Reply:
Circular deps between what and what?